x-pack/plugins/security_solution/public/plugin.tsx

```diff
@@ -355,112 +339,132 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
-        timelinesSubPlugin,
-        managementSubPlugin,
-      },
-      configIndexPatterns,
-    ] = await Promise.all([
-      this.downloadAssets(),
-      startPlugins.data.indexPatterns.getIdsWithTitle(),
-      this.downloadSubPlugins(),
-      startPlugins.data.search
-        .search<IndexFieldsStrategyRequest, IndexFieldsStrategyResponse>(
-          { indices: defaultIndicesName, onlyCheckIfIndicesExist: false },
-          {
-            strategy: 'securitySolutionIndexFields',
-          }
-        )
-        .toPromise(),
-    ]);
-
-    const { apolloClient } = composeLibs(coreStart);
-    const appLibs: AppObservableLibs = { apolloClient, kibana: coreStart };
-    const libs$ = new BehaviorSubject(appLibs);
-
-    const detectionsStart = detectionsSubPlugin.start(storage);
-    const hostsStart = hostsSubPlugin.start(storage);
-    const networkStart = networkSubPlugin.start(storage);
-    const timelinesStart = timelinesSubPlugin.start();
-    const managementSubPluginStart = managementSubPlugin.start(coreStart, startPlugins);
-
-    const timelineInitialState = {
-      timeline: {
-        ...timelinesStart.store.initialState.timeline!,
-        timelineById: {
-          ...timelinesStart.store.initialState.timeline!.timelineById,
-          ...detectionsStart.storageTimelines!.timelineById,
-          ...hostsStart.storageTimelines!.timelineById,
-          ...networkStart.storageTimelines!.timelineById,
+  /**
+   * Lazily instantiate a `SecurityAppStore`. We lazily instantiate this because it requests large dynamic imports. We instantiate it once because each subPlugin needs to share the same reference.
+   */
+  private async store(coreStart: CoreStart, startPlugins: StartPlugins): Promise<SecurityAppStore> {
+    if (!this._store) {
+      const defaultIndicesName = coreStart.uiSettings.get(DEFAULT_INDEX_KEY);
+      const [
+        { composeLibs, createStore, createInitialState },
+        kibanaIndexPatterns,
+        {
+          detections: detectionsSubPlugin,
+          hosts: hostsSubPlugin,
+          network: networkSubPlugin,
+          timelines: timelinesSubPlugin,
+          management: managementSubPlugin,
         },
-      },
-    };
+        configIndexPatterns,
+      ] = await Promise.all([
+        this.lazyApplicationDependencies(),
+        startPlugins.data.indexPatterns.getIdsWithTitle(),
+        this.subPlugins(),
+        startPlugins.data.search
+          .search<IndexFieldsStrategyRequest, IndexFieldsStrategyResponse>(
+            { indices: defaultIndicesName, onlyCheckIfIndicesExist: false },
+            {
+              strategy: 'securitySolutionIndexFields',
+            }
+          )
+          .toPromise(),
+      ]);
+
+      const { apolloClient } = composeLibs(coreStart);
+      const appLibs: AppObservableLibs = { apolloClient, kibana: coreStart };
+      const libs$ = new BehaviorSubject(appLibs);
+
+      const detectionsStart = detectionsSubPlugin.start(this.storage);
```

Comment:
`start` is called on most of these subPlugins more than once. That seems like an issue. I'm going to look into this a bit more.